File: algorithms/easy/maximum_value_of_a_string_in_an_array.ex
defmodule Solution do
  @spec maximum_value(strs :: [String.t()]) :: integer
  def maximum_value(strs) do
    strs
    |> Enum.map(fn s ->
      try do
        String.to_integer(s)
      rescue
        ArgumentError -> String.length(s)
      end
    end)
    |> Enum.max()
  end
end

File: algorithms/easy/maximum_value_of_a_string_in_an_array.cpp
class Solution {
public:
    int maximumValue(vector<string>& strs) {
      int ans = 0;
      for (auto& s : strs) {
        int i = 0;
        try {
          size_t pos{};
          i = stoi(s, &pos);
          if (pos != s.size()) i = s.size();
        } catch(invalid_argument const& ex) {
          i = s.size();
        }
        ans = max(ans, i);
      }
      return ans;
    }
};
